src/screens/NewsListScreen.js

```diff
@@ -81,24 +110,19 @@ const mapStateToProps = state => {
     isLoading: state.isLoading,
     articles: state.articles,
     page: state.page,
+    isDataValid: state.valid,
   };
 };
 
-const mapDispatchToProps = (dispatch, ownProps) => {
+const mapDispatchToProps = dispatch => {
   return {
     onRefresh: () => {
-      if (ownProps.isLoading) {
-        return;
-      }
-      dispatch(invalidateNews());
-      dispatch(loadArticles());
+      dispatch(invalidateArticles());
     },
     onLoadMore: () => {
-      if (ownProps.isLoading) {
-        return;
-      }
       dispatch(loadArticles());
     },
+    dispatch,
```

Comment:
Dont pass it to props, instead add `loadArticles: () => dispatch(loadArticles())`